pkg/tagger/tagger.go

```diff
@@ -132,6 +112,81 @@ func (t *Tagger) run() error {
 	}
 }
 
+// startCollectors iterates over the listener candidates and tries initializing them.
+// If the collector implements Retryer and return a FailWillRetry, we keep them in
+// the map and will retry at the next tick.
+func (t *Tagger) startCollectors() {
```

Comment:
The first part of `startCollectors` could be extract in a method like `getListenersCandidate`.
It also can returns an array of a specific struct with name, collector and mode.